src/Configuration/Config.AzureTableStorage/src/AzureTableStorageConfigurationProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureTableStorage
+{
+    internal class AzureTableStorageConfigurationProvider : ConfigurationProvider
+    {
+        private readonly ITableStore<ConfigurationEntry> configurationEntityStore;
+        private readonly string tableName;
+        private readonly string partitionKey;
+
+        public AzureTableStorageConfigurationProvider(ITableStore<ConfigurationEntry> configurationEntityStore, string tableName, string partitionKey)
+        {
+            if (string.IsNullOrWhiteSpace(tableName))
+            {
+                throw new ArgumentException($"{nameof(tableName)} cannot be null or white space.", nameof(tableName));
+            }
+
+            if (string.IsNullOrWhiteSpace(partitionKey))
+            {
+                throw new ArgumentException($"{nameof(partitionKey)} cannot be null or white space.", nameof(partitionKey));
+            }
+
+            this.configurationEntityStore = configurationEntityStore ?? throw new ArgumentNullException(nameof(configurationEntityStore));
+            this.tableName = tableName;
+            this.partitionKey = partitionKey;
+        }
+
+        public override void Load() => LoadAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+
+        private async Task LoadAsync()
+        {
+            var allItems = await configurationEntityStore.GetAllByPartitionKey(tableName, partitionKey);
```

Comment:
Do we need support for cancellation?

See for reference https://github.com/aspnet/Extensions/blob/7fd2ab9f058570ce241a19cb12bdf8d13151a861/src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs#L45

Reply:
Are you suggesting to add another extension overload for `AddAzureTableStorage` to allow passing cancellation token and just pass it along from `AzureTableStorageConfigurationProvider` onto the `GetAllByPartitionKey` call (and all subsequent table storage calls within the implementation)?

Or you are suggesting a `Reload` support like the `AzureKeyVaultConfigurationProvider`?

Reply:
FYI - I have already done the cancellation support for table storage calls.

Reply:
> Are you suggesting to add another extension overload

No, I just meant to make cancellation similar to how it is implemented in https://github.com/aspnet/Extensions/blob/7fd2ab9f058570ce241a19cb12bdf8d13151a861/src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationProvider.cs#L17 an pass the token to the table-queries.

Maybe someone from the team should comment about cancelaltion / reload.

Reply:
I've added the ability for consumers to pass cancellation. I'll wait for someone else to comment on the reload support.